packages/in_app_purchase/tizen/src/billing_manager.cc

```diff
@@ -0,0 +1,365 @@
+      app_id.c_str(), custom_id.c_str(), country_code.c_str(), page_num,
+      check_value.c_str(), billing_server_type_, OnPurchase, this);
+  if (!ret) {
+    LOG_ERROR("[BillingManager] service_billing_get_purchase_list failed.");
+    return false;
+  }
+  return true;
+}
+
+bool BillingManager::BuyItem(const flutter::EncodableMap *encodables) {
+  LOG_INFO("[BillingManager] Start buy item");
+
+  std::string pay_details =
+      GetRequiredArg<std::string>(encodables, "payDetails");
+  std::string app_id = GetRequiredArg<std::string>(encodables, "appId");
+
+  bool ret = BillingWrapper::GetInstance().service_billing_buyitem(
+      app_id.c_str(), ServerTypeToString(billing_server_type_).c_str(),
+      pay_details.c_str());
+  BillingWrapper::GetInstance().service_billing_set_buyitem_cb(OnBuyItem, this);
+  if (!ret) {
+    LOG_ERROR("[BillingManager] service_billing_buyitem failed.");
+    return false;
+  }
+  return true;
+}
+
+bool BillingManager::VerifyInvoice(const flutter::EncodableMap *encodables) {
+  LOG_INFO("[BillingManager] Start verify invoice");
+
+  std::string app_id = GetRequiredArg<std::string>(encodables, "appId");
+  std::string custom_id = GetRequiredArg<std::string>(encodables, "customId");
+  std::string invoice_id = GetRequiredArg<std::string>(encodables, "invoiceId");
+  std::string country_code =
+      GetRequiredArg<std::string>(encodables, "countryCode");
+
+  bool ret = BillingWrapper::GetInstance().service_billing_verify_invoice(
+      app_id.c_str(), custom_id.c_str(), invoice_id.c_str(),
+      country_code.c_str(), billing_server_type_, OnVerify, this);
+  if (!ret) {
+    LOG_ERROR("[BillingManager] service_billing_verify_invoice failed.");
+    return false;
+  }
+  return true;
+}
+
+BillingManager::BillingManager(flutter::PluginRegistrar *plugin_registrar) {
+  auto channel =
+      std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
+          plugin_registrar->messenger(),
+          "plugins.flutter.tizen.io/in_app_purchase",
+          &flutter::StandardMethodCodec::GetInstance());
+
+  channel->SetMethodCallHandler(
+      [this](const flutter::MethodCall<flutter::EncodableValue> &call,
+             std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>>
+                 result) { this->HandleMethodCall(call, std::move(result)); });
+  method_channel_ = std::move(channel);
+}
+
+void BillingManager::HandleMethodCall(
+    const flutter::MethodCall<flutter::EncodableValue> &method_call,
+    std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+  const auto *encodables =
+      std::get_if<flutter::EncodableMap>(method_call.arguments());
+  try {
+    const auto &method_name = method_call.method_name();
+    if (method_name == "getProductList") {
+      if (!encodables) {
+        result->Error(kInvalidArgument, "No arguments provided");
+        return;
+      }
+      if (GetProductList(encodables)) {
+        method_result_ = std::move(result);
```

Comment:
How to make sure the 'method_result_' is not changed by other function?

Reply:
Fixed!